task/push-build-file-oci-ta/0.1/push-build-file-oci-ta.yaml

```diff
@@ -0,0 +1,112 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: push-build-file-oci-ta
+  annotations:
+    tekton.dev/pipelines.minVersion: 0.12.1
+    tekton.dev/tags: image-build, appstudio
+  labels:
+    app.kubernetes.io/version: "0.1"
+    build.appstudio.redhat.com/build_type: docker
+spec:
+  description: Discover build file from source code and push it to registry
+    as an OCI artifact. The build file could be named Dockerfile or Containerfile,
+    which can be specified via parameter.
+  params:
+    - name: ARTIFACT_TYPE
+      description: Artifact type of the build file image.
+      type: string
+      default: application/vnd.konflux.dockerfile
+    - name: BUILD_FILE
+      description: Path to the build file.
+      type: string
+      default: ./Dockerfile
+    - name: CONTEXT
+      description: Path to the directory to use as context.
+      type: string
+      default: .
+    - name: IMAGE
+      description: The built binary image. The build file is pushed to the
+        same image repository alongside.
+      type: string
+    - name: IMAGE_DIGEST
+      description: The built binary image digest, which is used to construct
+        the tag of build file image.
+      type: string
+    - name: SOURCE_ARTIFACT
+      description: The Trusted Artifact URI pointing to the artifact with
+        the application source code.
+      type: string
+    - name: TAG_SUFFIX
+      description: Suffix of the build file image tag.
+      type: string
+      default: .dockerfile
+  results:
+    - name: IMAGE_URL
+      description: Digest-pinned image URL.
```

Comment:
Let's maybe use `IMAGE_REF`? Build tasks use `IMAGE_URL` as the digest-less repo url